src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -276,9 +274,64 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
 
   implicit val ClusterErrorFormat = jsonFormat3(ClusterError.apply)
 
-  implicit val ClusterFormat = jsonFormat22(Cluster.apply)
-
   implicit val DefaultLabelsFormat = jsonFormat6(DefaultLabels.apply)
 
   implicit val ClusterInitValuesFormat = jsonFormat20(ClusterInitValues.apply)
+
+  implicit object ClusterFormat extends RootJsonFormat[Cluster] {
+    override def read(json: JsValue): Cluster = {
+      json match {
+        case JsObject(fields: Map[String, JsValue]) =>
+          Cluster(
+            fields("id").convertTo[Long],
+            fields("clusterName").convertTo[ClusterName],
+            fields("googleProject").convertTo[GoogleProject],
+            fields("serviceAccountInfo").convertTo[ServiceAccountInfo],
+            DataprocInfo(fields("googleId").convertTo[Option[UUID]],
+                          fields("operationName").convertTo[Option[OperationName]],
+                          fields("stagingBucket").convertTo[Option[GcsBucketName]],
+                          fields("hostIp").convertTo[Option[IP]]),
+            AuditInfo(fields("creator").convertTo[WorkbenchEmail],
+                      fields("createdDate").convertTo[Instant],
+                      fields("destroyedDate").convertTo[Option[Instant]],
+                      fields("dateAccessed").convertTo[Instant]),
+            fields("machineConfig").convertTo[MachineConfig],
+            fields("clusterUrl").convertTo[URL],
+            fields("status").convertTo[ClusterStatus],
+            fields("labels").convertTo[LabelMap],
+            fields("jupyterExtensionUri").convertTo[Option[GcsPath]],
+            fields("jupyterUserScriptUri").convertTo[Option[GcsPath]],
+            fields("errors").convertTo[List[ClusterError]],
+            fields("instances").convertTo[Set[Instance]],
+            fields("userJupyterExtensionConfig").convertTo[Option[UserJupyterExtensionConfig]],
+            fields("autopauseThreshold").convertTo[Int])
+        case _ => deserializationError("Cluster expected")
+      }
+    }
+
+    override def write(obj: Cluster): JsValue = JsObject(
+      "id" -> obj.id.toJson,
+      "clusterName" -> obj.clusterName.toJson,
+      "googleId" -> obj.dataprocInfo.googleId.toJson,
```

Comment:
For absent optional fields, I think this will produce the JSON
```
{
  "googleId": null
}
```
when ideally we'd just not include the `googleId` key at all.

Can we filter out nulls from the list of fields? Something like:
```
      val allFields = List(
        "id" -> obj.id.toJson,
        "clusterName" -> obj.clusterName.toJson,
        "googleId" -> obj.dataprocInfo.googleId.toJson,
        "googleProject" -> obj.googleProject.toJson,
        "serviceAccountInfo" -> obj.serviceAccountInfo.toJson,
        "machineConfig" -> obj.machineConfig.toJson,
        "clusterUrl" -> obj.clusterUrl.toJson,
        "operationName" -> obj.dataprocInfo.operationName.toJson,
        "status" -> obj.status.toJson,
        "hostIp" -> obj.dataprocInfo.hostIp.toJson,
        "creator" -> obj.auditInfo.creator.toJson,
        "createdDate" -> obj.auditInfo.createdDate.toJson,
        "destroyedDate" -> obj.auditInfo.destroyedDate.toJson,
        "labels" -> obj.labels.toJson,
        "jupyterExtensionUri" -> obj.jupyterExtensionUri.toJson,
        "jupyterUserScriptUri" -> obj.jupyterUserScriptUri.toJson,
        "stagingBucket" -> obj.dataprocInfo.stagingBucket.toJson,
        "errors" -> obj.errors.toJson,
        "instances" -> obj.instances.toJson,
        "userJupyterExtensionConfig" -> obj.userJupyterExtensionConfig.toJson,
        "dateAccessed" -> obj.auditInfo.dateAccessed.toJson,
        "autopauseThreshold" -> obj.autopauseThreshold.toJson
      )

      val presentFields = allFields.filter(_._2 != JsNull)

      JsObject(presentFields:_*)
```